src/components/common/Menu/Menu.tsx

```diff
@@ -26,102 +30,133 @@ export const Menu = observer(() => {
         return null
     }
 
-    const links: Link[] = (
-        [
+    const getNavLinks = (): Link[] => {
+        const links: Link[] = [
             {
                 name: 'Search',
                 url: '/',
                 next: true,
+                active: router.asPath === '/',
```

Comment:
why to replace `map` with setting prop one by one? this introduces url prop duplicates in code

Reply:
I've replaced `map` because its a good general idea to avoid iterations when we can. We had 2x `.filter` and one `.map` before rendering the UI. If I also add the userProfileLinks, those will be itarateed 3 times as well. So we'll have 6 iterations before rendering the UI. Why not just creat them directly with the active prop in the constant definition?

Yes, url string will be duplicated for each entry, is that a bad thing? the code is readable and maintainable. 

Ultimately, this is subjective and a matter of personal preference. Both have pros and cons. (the previous solution with map being DRY effective)

Reply:
ok